cmd/api/main.go

```diff
@@ -2,14 +2,12 @@ package main
 
 import (
 	"omshub/core-api/internal/api"
-
-	"github.com/gin-gonic/gin"
 )
 
 func main() {
-	router := gin.Default()
-	router.GET("/", api.Index)
-	router.GET("/ping", api.Ping)
+	server := api.NewServer(api.Config{
+		Port: "3000",
+	})
 
-	_ = router.Run(":3000")
+	_ = server.Serve()
```

Comment:
We should probably dispatch a goroutine here to listen for OS signals, and invoke the `Shutdown` method when necessary. I'll make another issue for it, feel free to pick it up if you like!

Reply:
Done: https://github.com/omshub/core-api/issues/6